WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift

```diff
@@ -446,7 +448,22 @@ extension StoreStatsV4PeriodViewController: ChartViewDelegate {
     func chartValueSelected(_ chartView: ChartViewBase, entry: ChartDataEntry, highlight: Highlight) {
         let selectedIndex = Int(entry.x)
         updateUI(selectedBarIndex: selectedIndex)
-        usageTracksEventEmitter.interacted()
+
+        chartValueSelectedEventsSubject.send()
+    }
+
+    /// Observe `chartValueSelected` events and call `StoreStatsUsageTracksEventEmitter.interacted()` when
+    /// no similar events have been received after some time.
+    ///
+    /// We debounce it because there are just too many events received from `chartValueSelected()` when
+    /// the user holds and drags on the chart. Having too many events might skew the
+    /// `StoreStatsUsageTracksEventEmitter` algorithm.
+    private func observeChartValueSelectedEvents() {
```

Comment:
Let me know if this can be better! I was also not sure if it's worth it to create a unit test for this. I'm not sure how to begin testing debounce. Time travel? 😅 But let me know if you think otherwise!

Reply:
This is an interesting question! I did some search and it looks like we can mock the scheduler to test time-related signals from [this pointfree example](https://github.com/pointfreeco/combine-schedulers#testscheduler).

Though, I'm not sure if it's worth unit testing this use case since it might require some refactoring to extract this observation from the view controller. If this debounce logic is in the `StoreStatsUsageTracksEventEmitter` itself, it might be easier to unit test but this might not be worth the effort 😅